src/initialization/InitAmrCore.cpp

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2022 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Axel Huebl, Chad Mitchell, Ji Qiang
+ * License: BSD-3-Clause-LBNL
+ */
+#include "InitAmrCore.H"
+
+#include "initialization/InitAMReX.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_Box.H>
+#include <AMReX_CoordSys.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_RealBox.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+#include <AMReX_Vector.H>
+
+#include <stdexcept>
+
+
+namespace impactx::initialization
+{
+namespace details
+{
+
+    /** Set the initial physical simulation extent
+     *
+     * Sets to either a fake min:max box or to the user-provided extent.
+     * Controlled by geometry.dynamic_size.
+     *
+     * @return the RealBox in meters
+     */
+    amrex::RealBox
+    init_physical_domain ()
+    {
+        amrex::ParmParse pp_geometry("geometry");
+        bool dynamic_size = true;
+        pp_geometry.queryAdd("dynamic_size", dynamic_size);
+        amrex::Vector<amrex::Real> prob_lo(undefined_geometry_prob_lo.begin(), undefined_geometry_prob_lo.end());
+        amrex::Vector<amrex::Real> prob_hi(undefined_geometry_prob_hi.begin(), undefined_geometry_prob_hi.end());
+        if (dynamic_size)
+        {
+            pp_geometry.queryAdd("prob_lo", prob_lo);
+            pp_geometry.queryAdd("prob_hi", prob_hi);
+        }
+        else
+        {
+            pp_geometry.getarr("prob_lo", prob_lo);
+            pp_geometry.getarr("prob_hi", prob_hi);
+        }
+        amrex::RealBox rb(prob_lo.data(), prob_hi.data());
+        return rb;
+    }
+} // namespace details
+
+    AmrCoreData
+    init_amr_core ()
+    {
+        if (!amrex::Initialized())
```

Comment:
```suggestion
        // note: due to global state, it would be too early to use amrex::Abort() in these functions
        if (!amrex::Initialized())
```